index.d.ts

```diff
@@ -85,23 +85,27 @@ type StdioOptionsArray<IsSync extends boolean = boolean> = readonly [
 type StdioOptions<IsSync extends boolean = boolean> = BaseStdioOption | StdioOptionsArray<IsSync>;
 
 type DefaultEncodingOption = 'utf8';
-type BufferEncodingOption = 'buffer';
-type EncodingOption =
+type TextEncodingOption =
   | DefaultEncodingOption
   // eslint-disable-next-line unicorn/text-encoding-identifier-case
   | 'utf-8'
   | 'utf16le'
   | 'utf-16le'
   | 'ucs2'
-  | 'ucs-2'
-  | 'latin1'
-  | 'binary'
-  | 'ascii'
+  | 'ucs-2';
+type BufferEncodingOption = 'buffer';
+type BinaryEncodingOption =
+  | BufferEncodingOption
   | 'hex'
   | 'base64'
   | 'base64url'
-  | BufferEncodingOption
-  | undefined;
+  | 'latin1'
+  | 'binary'
+  | 'ascii';
```

Comment:
This PR still handles the encoding aliases, `null` and case-insensitivity. 
Per #925, I'll remove those in a follow-up PR.